src/network/networkd-state-file.c

```diff
@@ -217,6 +226,13 @@ int manager_save(Manager *m) {
         if (carrier_state >= LINK_CARRIER_STATE_ENSLAVED)
                 carrier_state = LINK_CARRIER_STATE_CARRIER;
 
+        if (links_online && !links_offline)
+                online_state = LINK_ONLINE_STATE_ONLINE;
+        else if (links_online && links_offline)
+                online_state = LINK_ONLINE_STATE_PARTIAL;
+        else if (links_offline)
+                online_state = LINK_ONLINE_STATE_OFFLINE;
```

Comment:
First, `link_online` and `links_offline` are integer, do not implicitly cast to boolean.

How about the following?
```
online_state = links_online > 0 ? (links_offline > 0 ? LINK_ONLINE_STATE_PARTIAL : LINK_ONLINE_STATE_ONLINE)
                                : (links_offline > 0 ? LINK_ONLINE_STATE_OFFLINE : _LINK_ONLINE_STATE_INVALID);
```